rp2040-hal/src/timer.rs

```diff
@@ -61,17 +59,32 @@ impl Timer {
     pub fn new(timer: TIMER, resets: &mut RESETS) -> Self {
         timer.reset_bring_down(resets);
         timer.reset_bring_up(resets);
-        Self { timer }
+        Self { _private: () }
+    }
+
+    // Safety: Must only be used for read-only operations
+    fn timer(&self) -> &crate::pac::timer::RegisterBlock {
+        unsafe { &*pac::TIMER::PTR }
     }
```

Comment:
Could we keep this as an unsafe function to force point of use to check on these requirements?

It seems too easy to forget in a subsequent one-liner PR.

Reply:
I inlined the 3 accesses instead. The difference between `unsafe { &*pac::TIMER::PTR }` and `unsafe { self.timer() }` is not big enough to justify a separate function.